docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,103 @@
+---
+id: slot-validation-actions
+sidebar_label: Slot Validation Actions
+title: Slot Validation Actions
+abstract: A slot validation action is a special type of custom action, designed to handle custom extraction and/or validation of slot values. This can be used to validate slots with predefined mappings or extract slots with custom mappings.
+---
+
+Rasa SDK provides three base classes that you can inherit from in order to implement custom slot extraction and
+validation logic:
+
+## `Action` class
+
+You can use this class to create a regular [custom action](./custom-actions.mdx). The base class will not prevent you
+from returning arbitrary events, however a slot validation action should only return `SlotSet` and `BotUttered` events.
+Any other event type will be filtered out by the default action [`action_extract_slots`](./default-actions.mdx#action_extract_slots).
+The name of the custom action must be specified in the `action` key of the relevant
+[custom slot mapping](./domain.mdx#custom-slot-mappings) in the domain.
+Note that the action name must be listed in the domain `actions` section too.
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` class in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+With this option, you do not need to specify the `action` key in the [custom slot mapping](./domain.mdx#custom-slot-mappings).
+
```

Comment:
There needs to be a N.B. note here that says, if you use this class to to do custom extraction, mapping conditions are ignored, in addition this suggestion since this will run even if a form is active:

```suggestion
You can extend the `ValidationAction` class in the Rasa SDK to define custom extraction and / or validation of slots that
can be set or updated whether a form is active or not.
With this option, you do not need to specify the `action` key in the [custom slot mapping](./domain.mdx#custom-slot-mappings), since the default action `action_extract_slots` runs `action_validate_slot_mappings` automatically. 

```

Reply:
Yup, will apply this suggestion. Do you mean bringing the existing N.B note further up after the first sentence?

Reply:
Ja, bring it up and phrase it stronger (I can't find the comment I may or may not have left on it sorry!)

Reply:
Done!